homeassistant/components/motioneye/camera.py

```diff
@@ -80,6 +121,23 @@ def camera_add(camera: dict[str, Any]) -> None:
 
     listen_for_new_cameras(hass, entry, camera_add)
 
+    platform = entity_platform.async_get_current_platform()
+    platform.async_register_entity_service(
+        SERVICE_SET_TEXT_OVERLAY,
+        SCHEMA_SERVICE_SET_TEXT,
```

Comment:
Don't pass a schema. Pass a dict with validators and the helper will generate an entity service schema.

Reply:
Given that the service schema uses `cv.has_at_least_one_key` ([here](https://github.com/home-assistant/core/pull/53411/files/e28f986657256d88d7bd8980c788f137afb8b54c#diff-d9ed771a84cb1b5703b1f1783b2e314c91f46c7697abf0c41799474ea4628f5cR86)), I'm not sure there's straightforward way to do what is suggested without losing that extra validation?

Some alternatives I could imagine:
 - Borrow code from `make_entity_service_schema` ([ref](https://github.com/home-assistant/core/blob/dev/homeassistant/helpers/config_validation.py#L876)) so this schema is created correctly.
 - Apply that validation step in the service handler instead of upfront.

Am I missing something, and/or what would you suggest in this instance?

Reply:
Yes, using `cv.make_entity_service_schema` in the schema should also work. 👍